File: Entities/ProblemProCon.cs
namespace MindYourMoodWeb.Entities
{
    public class ProblemProCon : Entity
    {
        public enum ProblemType
        {
            Pro,
            Con
        }

        //public int Id { get; set; }
        public ProblemIdea Idea { get; set; }
        public ProblemStep Step { get; set; }
        public Problem Problem { get; set; }
        public string ProConText { get; set; }
        public ProblemType Type { get; set; }
    }
}

File: ClientApp/src/app/_model/problemProCon.ts
import { Problem } from "./problem";
import { ProblemIdea } from "./problemIdea";
import { ProblemStep } from "./problemStep";

export interface ProblemProCon {
  id: number;
  idea: ProblemIdea;
  step: ProblemStep;
  problem: Problem;
  proConText: string;
  type: number;
}
